### ISSUE TITLE
Bug  in update mechanism

### ISSUE THREAD

**Issue by andreasrueedlinger** [typeorm/typeorm#1042]
When you try to update a json column, I detected that somehow the new value is not updated/written to DB. After some debugging I think that there is some bug in the update mechanism itself: in `EntityMetadataUtils` the paths are created for the update query, but it fails to create a path from eg. json objects or Date objects: the check `if (entity[key] instanceof Object && !(entity[key] instanceof Function)) {` thinks that the (hydrated, not persistent!!) value is an object with subpaths which is incorrect in this scenario.
Also I could not find any place where the hydrated value are converted back to persistent values in the update mechanism. Is that correct?
My use-case:
```ts
@Enity()
export class TestEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column('json')
  someDatesToPersist: Date[];
}
```

Am I doing something wrong or is this scenario not implemented?

**Comment by pleerock** [comment]
are you probably right, this bug issue introduced in 0.1.1 (0.1.1 should work fine). Its a bit problematic issue and I'll need to think how to resolve it properly and not to break embedded functionality

**Comment by andreasrueedlinger** [comment]
thanks for the fast response!

**Comment by andreasrueedlinger** [comment]
FYI:
* v0.1.1 skips the column in the update completely
* v0.1.0 includes the column in the update but does not call `preparePersistentValue` and thus tries to set an array of Date strings as column value

**Comment by pleerock** [comment]
Ive fixed it, this change will be released in `0.1.2` later this week. Thank you for you report and advices, they really helped me.
